server/src/main/java/io/crate/expression/symbol/GroupAndAggregateSemantics.java

```diff
@@ -76,6 +79,40 @@ public static void validate(List<Symbol> outputSymbols,
         }
     }
 
+    private static void validateGroupBySymbol(Symbol groupBy) {
+        groupBy.accept(SymbolTypeValidator.INSTANCE, null);
+    }
+
+    private static class SymbolTypeValidator extends SymbolVisitor<Void, Void> {
```

Comment:
Maybe we could make the names a bit specific. E.g. `ensurehasDefinedType` and `EnsureTypedGroupKey` or something like that.

`validator` and `validate` is always a bit vague in that it doesn't indicate what constraints or conditions need to be met for it to be valid